File: spikesToDelete/java/simulator/src/test/java/eu/route20/hft/simulation/SimulationRunnerTest.java
package eu.route20.hft.simulation;

import java.util.*;
import lombok.*;
import org.junit.*;
import org.mockito.*;

public class SimulationRunnerTest {
	private SimulationRunner runner;
	@Mock private Simulator s1;
	@Mock private Simulator s2;

	@Before public void setup() {
		MockitoAnnotations.initMocks(this);
		runner = new SimulationRunner();
	}

	@Test public void takes1toXSimulators() {
		val simulatorsWithOneSimulator = Arrays.asList(s1);
		runner.setSimulators(simulatorsWithOneSimulator);
		val simulatorsWithTwoSimulators = Arrays.asList(s1, s2);
		runner.setSimulators(simulatorsWithTwoSimulators);
	}

	@Ignore @Test public void runsAllSimulators() {
		val simulatorsWithTwoSimulators = Arrays.asList(s1, s2);
		runner.setSimulators(simulatorsWithTwoSimulators);
		runner.run();
		Mockito.verify(s1).doSimulation();
		Mockito.verify(s2).doSimulation();
	}

	@Ignore @Test public void runInParallel() {
		val simulatorsWithTwoSimulators = Arrays.asList(s1, s2);
		runner.setSimulators(simulatorsWithTwoSimulators);
		runner.run();
		InOrder order = Mockito.inOrder(s1, s2);
		order.verify(s1).doSimulation();
		order.verify(s2).doSimulation();

	}
}

File: scala/src/test/scala/SimulationRunnerTest.scala

package eu.route20.hft

import eu.route20.hft.test._
import org.scalatest._
import eu.route20.hft.simulation._

class SimulationRunnerTest extends BaseTest {

  "SimulationRunner" should "invoke simulator" in {
    val simulatorMock = mockFunction[Unit, Unit]
    def f = { simulatorMock() }
    val simulators = List(f _)
    simulatorMock expects ()
    val simulationRunner = new SimulationRunner(simulators)
    simulationRunner run
  }

  it should "invoke all configured simulators" in {
    val m1 = mockFunction[Unit, Unit]
    val m2 = mockFunction[Unit, Unit]
    def f1 = { m1() }
    def f2 = { m2() }
    val simulators = List(f1 _, f2 _)
    m1 expects ()
    m2 expects ()
    val runner = SimulationRunner(simulators)
    runner run
  }
}
